Lib/logging/handlers.py

```diff
@@ -827,11 +828,26 @@ def __init__(self, address=('localhost', SYSLOG_UDP_PORT),
             self.unixsocket = False
             if socktype is None:
                 socktype = socket.SOCK_DGRAM
-            self.socket = socket.socket(socket.AF_INET, socktype)
-            if socktype == socket.SOCK_STREAM:
-                self.socket.connect(address)
-            self.socktype = socktype
-        self.formatter = None
+            host, port = address
+            err = None
+            for res in socket.getaddrinfo(host, port, 0, socktype):
+                af, socktype, proto, cannonname, sa = res
```

Comment:
The canonical name isn't used, so could use `_` to bind it. Otherwise, use `canonical` as the name to bind to - `cannonname` is wrong.